examples/reown/src/providers/SyncedWalletProvider.tsx

```diff
@@ -0,0 +1,112 @@
+import { useSyncWagmiConfig } from '@lifi/wallet-management'
+import { ChainType, type ExtendedChain, useAvailableChains } from '@lifi/widget'
+import { BitcoinAdapter } from '@reown/appkit-adapter-bitcoin'
+import { SolanaAdapter } from '@reown/appkit-adapter-solana'
+import { WagmiAdapter } from '@reown/appkit-adapter-wagmi'
+import type { AppKitNetwork } from '@reown/appkit-common'
+import { bitcoin, solana } from '@reown/appkit/networks'
+import { type AppKit, createAppKit } from '@reown/appkit/react'
+import { PhantomWalletAdapter } from '@solana/wallet-adapter-wallets'
+import { useEffect, useRef } from 'react'
+import { WagmiProvider } from 'wagmi'
+import { metadata, projectId } from '../config/appkit'
+import { chainToAppKitNetworks, getChainImagesConfig } from '../utils/appkit'
+import { SolanaProvider, emitter } from './SolanaProvider'
+
+export function WalletProvider({
+  children,
+  chains,
+}: { children: React.ReactNode; chains: ExtendedChain[] }) {
+  const wagmi = useRef<WagmiAdapter | undefined>(undefined)
+  const modal = useRef<AppKit | undefined>(undefined)
+
+  const networks: [AppKitNetwork, ...AppKitNetwork[]] = [solana, bitcoin]
```

Comment:
Should we move this under the below `if` statement if it's not used outside of it anyway?